src/Airship.Net/Platforms/Android/Airship.cs

```diff
@@ -192,9 +192,87 @@ private static Features FeaturesFromUAFeatures(int uAFeatures)
             return features;
         }
 
-
         public IEnumerable<string> Tags => UAirship.Shared().Channel.Tags;
 
+        private class ResultCallback : Java.Lang.Object, IResultCallback
+        {
+            Action<Java.Lang.Object?> action;
+
+            internal ResultCallback(Action<Java.Lang.Object?> action)
+            {
+                this.action = action;
+            }
+
+            public void OnResult(Java.Lang.Object? result)
+            {
+                action.Invoke(result);
+            }
+        }
+
+        private List<string> CastHashSetToList(HashSet set)
+        {
+            var list = new List<string>();
+
+            var value = set.Iterator();
+            if (value is not null)
+            {
+                while (value.HasNext)
+                {
+                    var nextValue = (string?)value.Next();
+                    if (nextValue is not null)
+                    {
+                        list.Add(nextValue);
+                    }
+                }
+            }
+            return list;
+        }
+
+        public void FetchChannelSubscriptionLists(Action<List<string>> subscriptions)
+        {
+            PendingResult subscriptionsPendingResult = UAirship.Shared().Channel.FetchSubscriptionListsPendingResult();
+
+            subscriptionsPendingResult.AddResultCallback(new ResultCallback((result) =>
+            {
+                var list = new List<string>();
```

Comment:
I think we could maybe do something like this here to avoid creating an empty list, but it's probably not a big deal either way...

```csharp

subscriptionsPendingResult.AddResultCallback(new ResultCallback((result) 
{
    List<string> list = result switch
    {
        HashSet hashSet => CastHashSetToList(hashSet),
        _ => new List<string>(),
    }

    subscriptions(list);
}